data/en/test_multiple_piping.json

```diff
@@ -104,14 +104,57 @@
-                    "title": "Does <em>{{[answers['first-text'], answers['second-text']] | format_household_name}}</em> live at <em>{{answers['address-line-1']}}</em>",
+                    "title": {
+                        "text": "Does <em>{person}</em> live at <em>{address}</em>",
+                        "placeholders": [{
+                            "placeholder": "person",
+                            "transforms": [{
+                                "transform": "concatenate_list",
+                                "arguments": {
+                                    "list_to_concatenate": {
+                                        "source": "answers",
+                                        "identifier": ["first-text", "second-text"]
+                                    },
+                                    "delimiter": " "
+                                }
+                            }]
+                        }, {
+                            "placeholder": "address",
+                            "value": {
+                                "source": "answers",
+                                "identifier": "address-line-1"
+                            }
+                        }]
+                    },
                     "type": "General",
                     "answers": [{
                         "id": "multiple-piping-answer",
                         "type": "Radio",
                         "mandatory": true,
                         "options": [{
-                            "label": "{{[answers['building'], answers['address-line-1'], answers['address-line-2'], answers['address-line-3'], answers['town-city'], answers['county'], answers['postcode'], answers['country']] | concatenated_list}}",
+                            "label": {
+                                "text": "{address}",
+                                "placeholders": [{
+                                    "placeholder": "address",
+                                    "transforms": [{
+                                        "transform": "remove_empty",
+                                        "arguments": {
+                                            "list_to_filter": {
+                                                "source": "answers",
+                                                "identifier": ["building", "address-line-1", "address-line-2", "address-line-3", "town-city", "county", "postcode", "country"]
+                                            }
+                                        }
+                                    }, {
+                                        "transform": "concatenate_list",
+                                        "arguments": {
+                                            "list_to_concatenate": {
+                                                "source": "previous_transform"
+                                            },
+                                            "delimiter": ", "
```

Comment:
We could make `concatenate_list` remove empty items before concatenation. I can't think of a scenario when we'd want empty items in a concatenated list.

Reply:
I did start doing that originally but then decided that all of the other functions were doing 1 thing and didn't want to get into the state we had before with the jinja filters

Reply:
I understand that, but concatenate_list should never include empty items. I'd assume that it would filter them out without needing to remember to include an extra transform every time.

Reply:
Already fixed